src/cclib/parser/qchemparser.py

```diff
@@ -292,6 +297,12 @@ def generate_formula_histogram(self):
     def extract(self, inputfile, line):
         """Extract information from the file object inputfile."""
 
+        wfn_method = ['HF', 'MP2', 'RI-MP2', 'LOCAL_MP2', 'MP4', 'CCD', 'CCSD', \
```

Comment:
Since `extract()` is called for every line in the file, we should avoid redefining this. Can you place it as an attribute (`self.wfn_method = [...]`) under `def before_parsing(self)`?